.circleci/config.yml

```diff
@@ -1,8 +1,174 @@
 version: 2.1
+references:
+  npm_cache_key: &npm_cache_key
+                   v1-dependency-npm-{{ checksum "package-lock.json" }}
+
+
+executors:
+  study-executor:
+    docker:
+      - image: broadinstitute/study-server-build:02-14-2020A
+    working_directory: ~/repo
+
+commands:
+
+  build-and-deploy-study:
+    description: "Build and deploy an angular study website"
+    parameters:
+      study_key:
+        type: string
+      study_guid:
+        type: string
+    steps:
+      - checkout:
+          path: ~/repo
+      - run:
+          name: <<parameters.study_key>> update interpolated env vars
+          command: |
+            echo 'export ANGULAR_PROJECT_NAME=ddp-<<parameters.study_key>>' >> $BASH_ENV
+            echo 'export ANGULAR_PROJECT_DIR_PATH=projects/ddp-<<parameters.study_key>>' >> $BASH_ENV
+            # A hurried solution. I am sure there is a better way to get the vault server address. Perhaps we cook it into Dockerfile?
+            echo 'export VAULT_ADDR=`grep "address" /etc/consul-template/config/config.json  | grep -v localhost | grep -Eo "https.*[^\"]"`' >> $BASH_ENV
+            source $BASH_ENV
+      - run:
+          name: <<parameters.study_key>> show pipeline vars
+          command: |
+            echo "pipeline.id << pipeline.id >>"
+            echo "pipeline.number	 << pipeline.number	 >>"
+            echo "pipeline.project.git_url << pipeline.project.git_url >>"
+            echo "pipeline.project.type << pipeline.project.type >>"
+            echo "pipeline.git.tag << pipeline.git.tag >>"
+            echo "pipeline.git.branch << pipeline.git.branch >>"
+            echo "pipeline.git.revision << pipeline.git.revision >>"
+            echo "pipeline.git.base_revision << pipeline.git.base_revision >>"
+      - run:
+          name: <<parameters.study_key>> determine deployment environment
+          command: |
+            case "<< pipeline.git.branch >>" in
+              develop)
+                DEPLOY_ENV=dev
+              ;;
```

Comment:
I am guessing that excluding the Angular flags in dev was intentional. 
I think we need them on dev. Problem is that sometimes code runs without the --aot flag but does not build with it.
Since locally we typically just do the `ng serve`, building with --aot flag on dev is a chance to catch this problem.

Reply:
@bskinner had suggested that we keep these flags off when doing build-on-commit, but have a cron-based dev build that runs with them enabled to keep us honest.  Should we do `--aot` always, but not `--prod` on dev?